docs/core/compatibility/jit/9.0/fp-to-integer.md

```diff
@@ -1,11 +1,13 @@
 ---
-title: "Floating point to integer conversions are saturating"
-description: Learn about the breaking change in .NET 9 where floating point to integer conversions have saturating behavior.
+title: "Floating point-to-integer conversions are saturating"
+description: Learn about the breaking change in .NET 9 where floating point-to-integer conversions have saturating behavior.
 ms.date: 09/03/2024
 ---
-# Floating point to integer conversions are saturating
+# Floating point-to-integer conversions are saturating
 
-Floating point to integer conversions now have *saturating* behavior on x86 and x64 machines. Saturating behavior means that if the converted value is too small or large for target type, the value is set to the minimum or maximum value, respectively, for that type.
+Floating point-to-integer conversions now have *saturating* behavior on x86 and x64 machines. Saturating behavior means that if the converted value is too small or large for target type, the value is set to the minimum or maximum value, respectively, for that type.
+
+Additionally, the `ConvertToIntegerNative<TInteger>` methods on [Single](xref:System.Single.ConvertToIntegerNative`1(System.Single)), [Double](xref:System.Double.ConvertToIntegerNative`1(System.Double)), and [Half](xref:System.Half.ConvertToIntegerNative`1(System.Half)) now have *platform-specific* behavior. This behavior is not guaranteed to match the previous behavior (which was already non-deterministic), but rather does whatever is most efficient for the native platform. In most cases, however, the behavior matches the previous behavior.
```

Comment:
I might have explained this a bit poorly in my last comment. These APIs are new in .NET 9, so the wording here may be slightly confusing.

The new behavior that this general breaking change doc is discussing can come with overhead (negatively impacts performance) that is undesirable to some scenarios. So the main intent of these new APIs is to give users access to `(TInteger)val` that is `fast` but doesn't guarantee the behavior for values that are outside the representable range of the `TInteger` type (so for `(int)val`, that means `float.ConvertToIntegerNative<int>(val)` doesn't guarantee the exact result if the infinitely precise result was greater than `int.MaxValue` or less than `int.MinValue`).

Because it aims to be `fast`, it can typically be used to get the behavior that existed prior to this breaking change, but it isn't a strict guarantee. If users need a `strict` guarantee, then they can utilize the platform specific hardware intrinsics such as `Sse.ConvertToInt32(Vector128.CreateScalar(val))` to handle `(int)val` for `float`. This requires checking `if (Sse.IsSupported)` prior to use. -- Such behavior can be notably tricky, however, because other target platforms (such Arm64) already produce different results (they always saturate). So the general recommendation is to use the new behavior so that it is deterministic or to use the new `ConvertToIntegerNative` APIs if the performance impact is unacceptable and the non-determinism for out of range values is acceptable.

Reply:
Ahh, got it, thanks. I'll update.